### PR TITLE
gh-111031: Check more files in `test_tokenize`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-111031

### CODE REVIEW

**Review by AlexWaygood — CHANGES_REQUESTED** [review]
By default, this test only tests a random sample of ten test files:

https://github.com/python/cpython/blob/baefbb21d91db2d950706737a6ebee9b2eff5c2d/Lib/test/test_tokenize.py#L1917-L1918

You have to pass `-ucpu` to run the test on all test files. If I run `python -m test test_tokenize -ucpu` with your PR branch, the test fails:

```pytb
======================================================================
ERROR: test_random_files (test.test_tokenize.TestRoundtrip.test_random_files) (file='C:\\Users\\alexw\\coding\\cpython\\Lib\\test\\test_fstring.py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\alexw\coding\cpython\Lib\test\test_tokenize.py", line 1915, in test_random_files
    self.check_roundtrip(f)
  File "C:\Users\alexw\coding\cpython\Lib\test\test_tokenize.py", line 1812, in check_roundtrip
    tokens2_from2 = [tok[:2] for tok in tokenize.tokenize(readline2)]
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\cpython\Lib\tokenize.py", line 445, in tokenize
    yield from _generate_tokens_from_c_tokenizer(rl_gen.__next__, encoding, extra_tokens=True)
  File "C:\Users\alexw\coding\cpython\Lib\tokenize.py", line 541, in _generate_tokens_from_c_tokenizer
    raise TokenError(msg, (e.lineno, e.offset)) from None
tokenize.TokenError: ('unterminated string literal (detected at line 708)', (708, 40))
```

File: Lib/test/test_tokenize.py [hunk 1]
```diff
@@ -1908,9 +1908,6 @@ def test_random_files(self):
 
         testfiles.remove(os.path.join(tempdir, "test_unicode_identifiers.py"))
```
Comment (AlexWaygood) [inline Lib/test/test_tokenize.py]: On the other hand, it looks like we _can_ remove this line (and the five lines immediately above it). Now that the tokenizer no longer uses regular expressions, it seems it's able to tokenize this file fine

File: Lib/test/test_tokenize.py [hunk 2]
```diff
@@ -1908,9 +1908,6 @@ def test_random_files(self):
...
 
-        # TODO: Remove this once we can untokenize PEP 701 syntax
-        testfiles.remove(os.path.join(tempdir, "test_fstring.py"))
-
         for f in ('buffer', 'builtin', 'fileio', 'os', 'platform', 'sys'):
             testfiles.remove(os.path.join(tempdir, "test_%s.py") % f)
```
Comment (AlexWaygood) [inline Lib/test/test_tokenize.py]: These also all seem to pass, though I'm not sure why they're currently removed (there's no comment indicating why)

File: Lib/test/test_tokenize.py (hunk 1 above)
Comment (pablogsal) [reply Lib/test/test_tokenize.py]: Let's remove this as well

File: Lib/test/test_tokenize.py (hunk 2 above)
Comment (pablogsal) [reply Lib/test/test_tokenize.py]: Maybe it was due to time concerns, but I don't think these files are particularly bad. If it passes now, let's remove the line and let's run the buildbots just to be sure

**Comment by sobolevn** [comment]
Hm, indeed. Let's convert this PR to remove other stuff (I planned to create 2 PRs anyway for this).

> These also all seem to pass, though I'm not sure why they're currently removed (there's no comment indicating why)

https://github.com/python/cpython/commit/5f6fa826172c102eed88965d520e3937f6f1fb84#diff-b07c4f10c75a22e102f114227a0a9b7b763ddda7a330109994e69b29540ea9bbL583-L586

> Maybe it was due to time concerns

Before:

```
» ./python.exe -m test -ucpu test_tokenize -m test_random_files
Using random seed 454459699
0:00:00 load avg: 1.23 Run 1 test sequentially
0:00:00 load avg: 1.23 [1/1] test_tokenize

== Tests result: SUCCESS ==

1 test OK.

Total duration: 17.4 sec
Total tests: run=1 (filtered)
Total test files: run=1/1 (filtered)
Result: SUCCESS
```

After (+1 sec):

```
» ./python.exe -m test -ucpu test_tokenize -m test_random_files  
Using random seed 1054419542
0:00:00 load avg: 1.28 Run 1 test sequentially
0:00:00 load avg: 1.28 [1/1] test_tokenize

== Tests result: SUCCESS ==

1 test OK.

Total duration: 18.8 sec
Total tests: run=1 (filtered)
Total test files: run=1/1 (filtered)
Result: SUCCESS
```

**Review by sunmy2019 — APPROVED** [review]
Yeah, this is now correct. We still cannot untokenize `test_fstring.py` yet.

**Review by AlexWaygood — APPROVED** [review]
Just one buildbot failure, and test_tokenize passed on that buildbot. Thanks!

**Comment by AlexWaygood** [comment]
I'll defer to Pablo and Lys, but I feel we probably shouldn't backport this to 3.11, since the `tokenize`module is implemented completely differently in Python 3.11. 3.12 should be fine, though.

**Comment by lysnikolaou** [comment]
> I'll defer to Pablo and Lys, but I feel we probably shouldn't backport this to 3.11, since the `tokenize`module is implemented completely differently in Python 3.11. 3.12 should be fine, though.

Yup, 3.12 should be fine, but not 3.11.

**Pull request merged**